mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/mail/JpaMailboxMapperTest.java

```diff
@@ -49,4 +56,35 @@ protected MailboxId generateId() {
     void cleanUp() {
         JPA_TEST_CLUSTER.clear(JPAMailboxFixture.MAILBOX_TABLE_NAMES);
     }
+
+    @Test
+    void invalidUidValidityShouldBeSanitized() throws Exception {
+        EntityManager entityManager = JPA_TEST_CLUSTER.getEntityManagerFactory().createEntityManager();
+
+        entityManager.getTransaction().begin();
+        JPAMailbox jpaMailbox = new JPAMailbox(benwaInboxPath, -1L);// set an invalid uid validity
+        jpaMailbox.setUidValidity(-1L);
+        entityManager.persist(jpaMailbox);
+        entityManager.getTransaction().commit();
+
+        Mailbox readMailbox = mailboxMapper.findMailboxByPath(benwaInboxPath);
+
+        assertThat(readMailbox.getUidValidity().isValid()).isTrue();
+    }
+
+    @Test
+    void uidValiditySanitizingShouldPersistTheSanitizedUidValidity() throws Exception {
+        EntityManager entityManager = JPA_TEST_CLUSTER.getEntityManagerFactory().createEntityManager();
+
+        entityManager.getTransaction().begin();
+        JPAMailbox jpaMailbox = new JPAMailbox(benwaInboxPath, -1L);// set an invalid uid validity
+        jpaMailbox.setUidValidity(-1L);
+        entityManager.persist(jpaMailbox);
+        entityManager.getTransaction().commit();
+
+        Mailbox readMailbox1 = mailboxMapper.findMailboxByPath(benwaInboxPath);
+        Mailbox readMailbox2 = mailboxMapper.findMailboxByPath(benwaInboxPath);
+
+        assertThat(readMailbox1.getUidValidity()).isEqualTo(readMailbox2.getUidValidity());
```

Comment:
Can you explain me this test? I don't see how it's relevant to check that UidValidity is the same on two same `findMailboxByPath` action after persisting with JPA?

Reply:
It's to be sure that we did persist the sanitized value ^^